experimental/packages/opentelemetry-instrumentation-grpc/src/grpc-js/clientUtils.ts

```diff
@@ -135,90 +182,68 @@ export function makeGrpcClientRemoteCall(
-        });
-
-        endSpan();
-      });
-
-      call.on('status', (status: SpanStatus) => {
-        if (call[CALL_SPAN_ENDED]) {
-          return;
-        }
-        call[CALL_SPAN_ENDED] = true;
-
-        span.setStatus(_grpcStatusCodeToSpanStatus(status.code));
-        span.setAttribute(SemanticAttributes.RPC_GRPC_STATUS_CODE, status.code);
-
-        endSpan();
-      });
+      patchResponseStreamEvents(span, call);
     }
     return call;
   };
 }
 
-/**
- * Returns the metadata argument from user provided arguments (`args`)
- */
-export function getMetadata(
-  this: GrpcJsInstrumentation,
-  grpcClient: typeof grpcJs,
-  original: GrpcClientFunc,
+export function getMetadataIndex(
   args: Array<unknown | grpcJs.Metadata>
-): grpcJs.Metadata {
-  let metadata: grpcJs.Metadata;
-
+): number {
   // This finds an instance of Metadata among the arguments.
   // A possible issue that could occur is if the 'options' parameter from
   // the user contains an '_internal_repr' as well as a 'getMap' function,
   // but this is an extremely rare case.
-  let metadataIndex = args.findIndex((arg: unknown | grpcJs.Metadata) => {
+  return args.findIndex((arg: unknown | grpcJs.Metadata) => {
     return (
       arg &&
       typeof arg === 'object' &&
       (arg as grpcJs.Metadata)['internalRepr'] && // changed from _internal_repr in grpc --> @grpc/grpc-js https://github.com/grpc/grpc-node/blob/95289edcaf36979cccf12797cc27335da8d01f03/packages/grpc-js/src/metadata.ts#L88
       typeof (arg as grpcJs.Metadata).getMap === 'function'
     );
   });
+}
+
+/**
+ * Returns the metadata argument from user provided arguments (`args`)
+ * If no metadata is provided in `args`: adds empty metadata to `args` and returns that empty metadata
+ */
+export function extractMetadataOrSplice(
+  grpcLib: typeof grpcJs,
+  args: Array<unknown | grpcJs.Metadata>,
+  spliceIndex: number
+) {
+  let metadata: grpcJs.Metadata;
+  const metadataIndex = getMetadataIndex(args);
   if (metadataIndex === -1) {
-    metadata = new grpcClient.Metadata();
-    if (!original.requestStream) {
-      // unary or server stream
-      metadataIndex = 1;
-    } else {
-      // client stream or bidi
-      metadataIndex = 0;
-    }
-    args.splice(metadataIndex, 0, metadata);
+    // Create metadata if it does not exist
+    metadata = new grpcLib.Metadata();
+    args.splice(spliceIndex, 0, metadata);
   } else {
     metadata = args[metadataIndex] as grpcJs.Metadata;
   }
   return metadata;
 }
 
+/**
+ * Returns the metadata argument from user provided arguments (`args`)
+ * Adds empty metadata to arguments if the default is used.
+ */
+export function extractMetadataOrSpliceDefault(
+  grpcClient: typeof grpcJs,
+  original: GrpcClientFunc,
+  args: Array<unknown | grpcJs.Metadata>
+): grpcJs.Metadata {
+  if (!original.requestStream) {
+    // unary or server stream
+    return extractMetadataOrSplice(grpcClient, args, 1);
```

Comment:
maybe use `extractMetadataOrSplice(grpcClient, args, original.requestStream ? 0 : 1);`?

Reply:
Done in [3b192ba](https://github.com/open-telemetry/opentelemetry-js/pull/3804/commits/3b192ba02372802617d438347ec12c47b439d3eb) :slightly_smiling_face: